### ISSUE TITLE
Invalid module import error (no-name-in-module)

### ISSUE THREAD

**Issue by martinnovak** [pylint-dev/pylint#9387]
### Bug description

When you have a module utils/collections.py and in other folder, you have utils2/main.py, pytlint reports

`util2/main.py:1:0: E0611: No name 'namedtuple' in module 'collections' (no-name-in-module)`

Whereas when running python, there is no problem with running

`python3 util2/main.py`

I guess this is false positive error.

I created a simple repo where you can test the problem https://github.com/martinnovak/pylint-bug/

### Command used

```shell
pylint .
```

### Pylint output

```python
(pylint-bug) mango:~/pylint-bug$ pylint .
************* Module main
util2/main.py:1:0: E0611: No name 'namedtuple' in module 'collections' (no-name-in-module)
```

### Expected behavior

I expect to run with no error.

### Pylint version

```shell
(pylint-bug) mango:~/pylint-bug$ pylint --version
pylint 3.0.3
astroid 3.0.2
Python 3.11.7 (main, Dec  4 2023, 18:10:11) [Clang 15.0.0 (clang-1500.1.0.2.5)]
```

### OS / Environment

macOS 14.3 (23D56)

**Comment by jacobtylerwalls** [comment]
Duplicate of #6535
